wallet/src/ui/app/components/network-switch/NetworkSelector.tsx

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import cl from 'classnames';
+import { useMemo, useCallback } from 'react';
+
+import { API_ENV_TO_INFO, API_ENV } from '_app/ApiProvider';
+import BsIcon from '_components/bs-icon';
+import { useAppSelector, useAppDispatch } from '_hooks';
+import {
+    setApiEnv,
+    changeRPCNetwork,
+    setNetworkSelector,
+} from '_redux/slices/app';
+import { getTransactionsByAddress } from '_redux/slices/txresults';
+import store from '_store';
+
+import st from './Network.module.scss';
+
+const NetworkSelector = () => {
+    const selectedApiEnv = useAppSelector(({ app }) => app.apiEnv);
+    const dispatch = useAppDispatch();
+    const networkList: any = API_ENV_TO_INFO;
+    const netWorks = useMemo(
+        () =>
+            Object.keys(networkList).map((itm: string) => ({
+                style: { color: networkList[itm].color },
+                ...networkList[itm],
+                networkName: itm,
+            })),
+        [networkList]
+    );
+
+    const changeNetwork = useCallback(
+        (networkName: string) => () => {
+            const apiEnv = API_ENV[networkName as keyof typeof API_ENV];
+            dispatch(setNetworkSelector(true));
```

Comment:
This feels like it should be a local state to the 'dropdown' component instead of storing it to redux. Maybe we can refactor it to have a generic dropdown component (maybe in another MR)

Reply:
Not sure how this will work. Create a local state, listen to the changes in the local state and dispatch an action whenever the is a change?